requirements.txt

```diff
@@ -1,4 +1,26 @@
-mysql-connector-python
-ops==2.21.0
-requests==2.32.3
-pydantic>=1,<2
+cosl==0.1.1
+mysql-connector-python==9.3.0
+ops==2.22.0
+pydantic==1.10.22
+requests==2.32.4
+
+# indirect dependencies
+anyio==4.9.0
+certifi==2025.6.15
+charset-normalizer==3.4.2
+exceptiongroup==1.3.0
+h11==0.16.0
+httpcore==1.0.9
+httpx==0.28.1
+idna==3.10
+importlib_metadata==8.7.0
+lightkube==0.17.2
+lightkube-models==1.33.1.8
+opentelemetry-api==1.34.1
+PyYAML==6.0.2
+sniffio==1.3.1
+tenacity==9.1.2
+typing_extensions==4.14.0
+urllib3==2.4.0
+websocket-client==1.8.0
+zipp==3.23.0
```

Comment:
For my own learning, how may I track the indirect dependencies?

Reply:
This is for the new feature in charmcraft called `charm-strict-dependencies`. As the dependencies for the WordPress charm are currently split between `charmcraft.yaml` and `requirements.txt` because some are binary packages. I am trying to use `charm-strict-dependencies` to consolidate all dependencies into `requirements.txt`.

For more information:
https://discourse.charmhub.io/t/upcoming-charmcraft-changes-and-a-call-for-testers/12175